### PR TITLE
Clean up Image

### PR DESCRIPTION
**Pull request by BlueCube3310**
This PR attempts to clean up the Image class by doing the following:
- Removing unnecessary comments,
- Grouping together similar parts of the code, which used to be scattered around the files,
- Moving some never used inline functions from public to private,
- Moving function definitions to the .cpp file.

### CODE REVIEW

File: core/io/image.h [hunk 1]
```diff
@@ -162,24 +214,26 @@ class Image : public Resource {
...
 
+	/* External VRAM decompression function pointers. */
+
 	static void (*_image_decompress_bc)(Image *);
 	static void (*_image_decompress_bptc)(Image *);
 	static void (*_image_decompress_etc1)(Image *);
 	static void (*_image_decompress_etc2)(Image *);
 	static void (*_image_decompress_astc)(Image *);
 
+	/* External packer function pointers. */
```
Comment (AThousandShips) [inline core/io/image.h:225]:
```suggestion
	// External packer function pointers.
```

File: core/io/image.h [hunk 2]
```diff
@@ -327,27 +344,14 @@ class Image : public Resource {
...
-	Image(int p_width, int p_height, bool p_use_mipmaps, Format p_format);
-	/**
-	 * import an image of a specific size and format from a pointer
-	 */
-	Image(int p_width, int p_height, bool p_mipmaps, Format p_format, const Vector<uint8_t> &p_data);
+	Image() = default; // Create an empty image.
+	Image(int p_width, int p_height, bool p_use_mipmaps, Format p_format); // Create an empty image of a specific size and format.
+	Image(int p_width, int p_height, bool p_mipmaps, Format p_format, const Vector<uint8_t> &p_data); // Import an image of a specific size and format from a byte vector.
+	Image(const uint8_t *p_mem_png_jpg, int p_len = -1); // Import either a png or jpg from a pointer.
+	Image(const char **p_xpm); // Import a XPM image.
```
Comment (AThousandShips) [inline core/io/image.h:351]:
```suggestion
	Image(const char **p_xpm); // Import an XPM image.
```

File: core/io/image.h [hunk 3]
```diff
@@ -190,13 +244,10 @@ class Image : public Resource {
...
-		height = p_image.height;
-		mipmaps = p_image.mipmaps;
-		data = p_image.data;
-	}
+	void _copy_internals_from(const Image &p_image);
+
+	_FORCE_INLINE_ Color _get_color_at_ofs(const uint8_t *ptr, uint32_t ofs) const;
+	_FORCE_INLINE_ void _set_color_at_ofs(uint8_t *ptr, uint32_t ofs, const Color &p_color);
 
 	_FORCE_INLINE_ void _get_mipmap_offset_and_size(int p_mipmap, int64_t &r_offset, int &r_width, int &r_height) const; //get where the mipmap begins in data
```
Comment (AThousandShips) [inline core/io/image.h:252]:
```suggestion
	_FORCE_INLINE_ void _get_mipmap_offset_and_size(int p_mipmap, int64_t &r_offset, int &r_width, int &r_height) const; // Get where the mipmap begins in data.
```
Missed one

**Review by AThousandShips — APPROVED** [review]
Looks good otherwise, I thin this is a worthwhile change even though noisy

**Review by clayjohn — APPROVED** [review]
Very nice cleanup!

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**